app/controllers/omniauth_controller.rb

```diff
@@ -100,6 +104,10 @@ def try_to_extract_user_uid
   rescue StandardError
     "unknown-provider-uid"
   end
+
+  def user_has_email_change_that_will_clash_with_existing_users
+    @user.changes[:email] && User.where(email: @user.changes[:email].last).any?
```

Comment:
having a look at the method `find_or_create_from_provider_data`, we are saving the user there before we get here, so we might have the user already with the new email and this might always return true if email changed and return the user we actually want to keep?

Reply:
If the email changed to one that doesn't clash, then the user will be saved successfully, and user.changes will be empty, as the new record has been persisted.
I'll add a feature spec to cover this scenario.